src/plots/cartesian/constraints.js

```diff
@@ -53,14 +55,14 @@ exports.handleConstraintDefaults = function(containerIn, containerOut, coerce, a
     // 'matches' wins over 'scaleanchor' (for now)
     var scaleanchor, scaleOpts;
 
-    if(!matches && containerIn.scaleanchor && !(containerOut.fixedrange && constrain !== 'domain')) {
+    if(!matches && (containerIn.scaleanchor || opts.scaleanchorDflt) && !(containerOut.fixedrange && constrain !== 'domain')) {
```

Comment:
This line is getting too long.
It would be nice to put conditions on multiple lines.

Reply:
Done in d71c3b345d2eecb7b3087c383f00f5f102b9c2f6